apiserver/src/apiserver/events.clj

```diff
@@ -0,0 +1,38 @@
+(ns apiserver.events
+  (:require
+   [clojure.tools.logging :as log]
+   [clojure.java.io :as io]
+   [ring.core.protocols :as p])
+  (:import
+   [com.rabbitmq.stream MessageHandler OffsetSpecification]))
+
+(defn ndaysago [n]
+  (- (System/currentTimeMillis) (-> (* n 24) (* 60) (* 60) (* 1000))))
+
+(defn events-handler
+  [{{:keys [environment name]} :stream-env}]
+  {:status 200
+   :headers {"content-type" "text/event-stream"
+             "transfer-encoding" "chunked"}
+   :body (reify p/StreamableResponseBody
+           (write-body-to-stream [_ _ output-stream]
+             (with-open [w (io/writer output-stream)]
+               (let [complete (promise)
+                     consumer (.. environment
+                                  consumerBuilder
+                                  (stream name)
+                                  (offset (OffsetSpecification/timestamp (ndaysago 7)))
```

Comment:
lets create the stream with the retention settings instead of setting an offset. that way the stream will be of a fixed size and not growing indefinitely.

Reply:
maybe both? I thought that a CI might want to show last seven days when opening the details of a pipeline. That's of most interest, but when the client asks for more historical runs it should be possible?

Reply:
Yeah my thoughts are XT is temporal and we use older version of things to show. And the pipeline logs are there. Events are more for real time reactions and not historical use. Not too sure what if people would prefer the events that led to a thing or the final state to debug.

Reply:
k8s for instance has a TTL for a events: https://medium.com/@andrew.kaczynski/kubernetes-events-how-to-keep-historical-data-of-your-cluster-835d685cc45 it expects us to see the history of the resources rather than events.

Reply:
ok, I see, found the retention setting now. but that means, when a client is requesting older data it is going through another endpoint, right?

Reply:
Yeah I think if the request has an older time stamp than the retention setting we just give out

Reply:
I think not having an infinite event storage should be better.